File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.go
func twoSum(numbers []int, target int) []int {
	left, right := 0, len(numbers) - 1

	for ; left < right; {
		sum := numbers[left] + numbers[right]

		if sum < target {
			left++
			continue
		}

		if sum > target {
			right--
			continue
		}

		return []int{left + 1, right + 1}
	}
	
	return []int{}
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int search(std::vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size() - 1;
        
        while (left <= right) {
            auto mid = left + (right - left) / 2;

            if (nums[mid] == target)
            {
                return mid;
            }

            if (nums[left] == target)
            {
                return left;
            }

            if (nums[right] == target)
            {
                return right;
            }

            if (target > nums[mid]) {
                if (nums[mid] > nums[left]) {
                    left = mid + 1;
                } else {
                    if (target > nums[left] )
                    {
                        right = mid - 1;
                    } else {
                        left = mid + 1;
                    }
                }
            } else {
                if (nums[mid] < nums[right]) {
                    right = mid - 1;
                } else {
                    if (target < nums[right] )
                    {
                        left = mid + 1;
                    } else {
                        right = mid - 1;
                    }
                }
            }
        }

        return -1;
    }
};
